engine/schema/resources/META-INF/db/schema-41000to41100.sql

```diff
@@ -19,8 +19,54 @@
 -- Schema upgrade from 4.10.0.0 to 4.11.0.0
 --;
 
+--;
+-- Stored procedure to do idempotent column add;
+--;
+DROP PROCEDURE IF EXISTS `cloud`.`IDEMPOTENT_ADD_COLUMN`;
```

Comment:
@ernjvr /cc @DaanHoogland @rafaelweingartner  - is it possible to refactor these methods out to a Java/sql file, and automatically include them in all sql files so that these procedures may be reused (somehow)? Or, perhaps explore migrating to something else that provides these abstractions, perhaps flywaydb etc?

Reply:
I would also rather migrate to something like flywaydb too. However, I think it would required quite more work than this one here. What do you think? I have done this analysis before, but I was not having much time, and the discussion died [1]. Maybe we could pick this topic up again?

[1] http://markmail.org/message/2dxtcnqx2euq5tl2

Reply:
@rafaelweingartner yes, we will have to make sure that everyone creating new scripts to add a new column, drop a foreign key or create/drop indexes, call these stored procedures, to ensure their SQL statements are idempotent. 
@rhtyd These 'drop procedure' statements are placed before each 'create procedure' statement only. So when the upgrade script runs the procedures are created but not dropped after the script completes. Therefore the stored procedures will still exist in the database. So they will be available (reusable) to future versions of schema-nn-nn.sql migration scripts. I think these 'drop/create procedure' statements also need to be added to the setup/db/create-schema.sql script to make them available to new cs installs, including the simulator.